nixos/modules/services/web-apps/nextcloud.nix

```diff
@@ -367,6 +369,33 @@ in {
         The nextcloud-occ program preconfigured to target this Nextcloud instance.
       '';
     };
+
+    extraOptions = mkOption {
+      type = jsonFormat.type;
+      default = {};
+      description = ''
+        Extra options which should be appended to nextcloud's config.php file.
+      '';
+      example = {
+        redis = {
+          host = "/run/redis/redis.sock";
+          port = 0;
+          dbindex = 0;
+          password = "secret";
+          timeout = 1.5;
+        };
+      };
+    };
+
+    secretFile = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Secret options which will be appended to nextcloud's config.php file (written as JSON, in the same
+        form as the <xref linkend="opt-services.nextcloud.extraOptions"/> option), for example
+        '{"redis":{"password":"secret","timeout"=2}}'.
```

Comment:
Tbh I don't really like that interface. Can you at least make sure that one can write
```
{
  "foo": "bar"
}
```

Also, I'm wondering if we want to do some kind of `file_get_contents`-logic for a single option specifically. And I don't really understand why `timeout` has to be here.

Reply:
this should already work.

The "timeout"=2 bit was a mistake (it don't think it should hurt to set it here, though using a "=" for that is definitely wrong since it's not valid json); I've removed it.